File: 0162_Find Peak Element/FindPeakElement_Iteration.java
class Solution {
    public int findPeakElement(int[] nums) {
        return findPeakElement(nums, 0, nums.length-1);
    }
    
    private int findPeakElement(int[] nums, int l , int r) {
        if(l == r) return l;
        int mid = (r-l)/2 + l;
        if(nums[mid] > nums[mid+1])
            return findPeakElement(nums, l , mid);
        else
            return findPeakElement(nums, mid + 1, r);
    }
}

File: 0162_Find Peak Element/FindPeakElement_Iteration.cs
public class Solution {
    public int FindPeakElement(int[] nums) {
        var l = 0;
        var r = nums.Length -1;
        while(l < r){
            var m = (r-l)/2 + l;
            // Find the first m which num[m] > num[m + 1]
            if(nums[m] > nums[m+1]){
                r = m;
            }else{
                l = m+1;
            }
        }
        
        return l;
    }
}